modules/provision/zdb.go

```diff
@@ -186,29 +191,21 @@ func zdbProvision(ctx context.Context, reservation *Reservation) (interface{}, e
 	slog.Info().IPAddr("ip", containerIP).Msg("container IP found")
 	addr := zdbConnectionURL(containerIP.String(), zdbPort)
 
-	slog.Debug().Str("addr", addr).Msg("connect to 0-db and create namespace")
-	zdbCl := zdb.New()
-	if err := zdbCl.Connect(addr); err != nil {
-		return nil, errors.Wrapf(err, "failed to connect to 0-db at %s", addr)
-	}
-	defer zdbCl.Close()
-
-	if err := zdbCl.CreateNamespace(nsID); err != nil {
-		return nil, errors.Wrapf(err, "failed to create namespace in 0-db at %s", addr)
-	}
-
-	if config.Password != "" {
-		if err := zdbCl.NamespaceSetPassword(nsID, config.Password); err != nil {
-			return nil, errors.Wrapf(err, "failed to set password namespace %s in 0-db at %s", nsID, addr)
-		}
-	}
-
-	if err := zdbCl.NamespaceSetPublic(nsID, config.Public); err != nil {
-		return nil, errors.Wrapf(err, "failed to make namespace %s public in 0-db at %s", nsID, addr)
+	netNS, err := namespace.GetByName(netNSName)
+	if err != nil {
+		return nil, err
 	}
-
-	if err := zdbCl.NamespaceSetSize(nsID, config.Size*gigabyte); err != nil {
-		return nil, errors.Wrapf(err, "failed to set size on namespace %s in 0-db at %s", nsID, addr)
+	defer netNS.Close()
+
+	if err := netNS.Do(func(_ ns.NetNS) error {
+		slog.Info().
+			Str("name", nsID).
+			Str("netns", netNSName).
+			Str("addr", addr).
+			Msg("create 0-db namespace")
+		return createZDBNamespace(addr, nsID, config)
```

Comment:
If you already are inside the zdb namespace, why do u need the address? the port should be deterministic, no ? unless all zdbs are running inside the same namespace. is that the case?

Reply:
Well I need to know to which IP I need to connect. 
And no, each 0-db container runs in their own network namespace